src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -455,10 +455,16 @@ export class AttachToDropdown extends SelectBox {
 	 **/
 	public async openConnectionDialog(useProfile: boolean = false): Promise<boolean> {
 		try {
+			//Get all providers to show all available connections in connection dialog
+			let providers = [];
+			providers = this.model.getApplicableConnectionProviderIds(this.model.clientSession.kernel.name);
```

Comment:
if you're setting this here, can just do

`let providers = this.model.`etc

Reply:
does `getApplicableConnectionProviderIds` return an empty array, or undefined?

Reply:
Yes, `getApplicableConnectionProviderIds` would return undefined if their arent any provider ids.